### ISSUE TITLE
XElement comparison is broken

### ISSUE THREAD

**Issue by ramankarman** [derekgreer/expectedObjects#32]
Hi.
Noticed strange behavior when trying to compare to objects with XElement property inside. Is it a bug? Can I avoid it somehow? The comparison works in 1.* version and doesn't in 2.*.

```
public class MyClass
{
    public System.Xml.Linq.XElement Prop { get; set; }
}

[TestFixture]
public class TestClass
{
    [Test]
    public void MyTest()
   {
            var instance1 = new MyClass { Prop = XElement.Parse("<root></root>") };
            var instance2 = new MyClass { Prop = XElement.Parse("<root></root>") };

            instance1
                .ToExpectedObject()
                .ShouldEqual(instance2);
    }
}
```

Exception message is 

```
Message: 
    System.InvalidOperationException : Sequence contains more than one matching element
  Stack Trace: 
    Enumerable.SingleOrDefault[TSource](IEnumerable`1 source, Func`2 predicate)
    <>c__DisplayClass8_0.<CompareProperties>b__0(PropertyInfo propertyInfo) line 76
    List`1.ForEach(Action`1 action)
    EqualityComparer.CompareProperties(Object expected, Object actual, Func`3 propertyComparison) line 74
    ClassComparisonStrategy.AreEqual(Object expected, Object actual, IComparisonContext comparisonContext) line 16
    EqualityComparer.Compare(Object expected, Object actual, String member, IWriter writer) line 190
    EqualityComparer.ReportEquality(Object expected, Object actual, String member) line 59
    ClassComparisonStrategy.CompareStandardProperty(PropertyInfo pi1, PropertyInfo pi2, Object expected, Object actual, IComparisonContext comparisonContext) line 105
    ClassComparisonStrategy.CompareProperty(PropertyInfo expectedPropertyInfo, PropertyInfo actualPropertyInfo, Object expected, Object actual, IComparisonContext comparisonContext, Boolean areEqual) line 49
    <>c__DisplayClass1_0.<AreEqual>b__0(PropertyInfo pi, PropertyInfo actualPropertyInfo) line 18
    <>c__DisplayClass8_0.<CompareProperties>b__0(PropertyInfo propertyInfo) line 76
    List`1.ForEach(Action`1 action)
    EqualityComparer.CompareProperties(Object expected, Object actual, Func`3 propertyComparison) line 74
    ClassComparisonStrategy.AreEqual(Object expected, Object actual, IComparisonContext comparisonContext) line 16
    EqualityComparer.Compare(Object expected, Object actual, String member, IWriter writer) line 190
    EqualityComparer.ReportEquality(Object expected, Object actual, String member) line 59
    EqualityComparer.AreEqual(Object expected, Object actual, IWriter writer, Boolean ignoreTypeInformation) line 122
    ExpectedObject.Equals(Object actual, IWriter writer, Boolean ignoreTypeInformation) line 25
    ShouldExtensions.ShouldEqual[T](ExpectedObject expected, T actual) line 13
    
```

**Comment by derekgreer** [comment]
This should be fixed in version 2.3.6 (see https://github.com/derekgreer/expectedObjects/blob/master/src/ExpectedObjects.Specs/XElementSpecs.cs)

**Comment by ramankarman** [comment]
Awesome
